source/common.ts

```diff
@@ -0,0 +1,46 @@
+import { createOwnCloudFileDetailProperty } from './helper';
+import { NextcloudClientInterface }         from './types';
+import { dirname, basename }                from 'path';
+import * as assert                          from 'assert';
+
+export async function getFileOrFolderCreator(fullPath) : Promise<string> {
+  const self: NextcloudClientInterface = this;
+
+  let result = null;
+
+  const baseFolder       = dirname(fullPath);
+  const fileOrFolderName = basename(fullPath);
+
+  try {
+    assert(await self.exists(baseFolder));
+    const folderFileDetails = await self.getFolderFileDetails(
+      baseFolder, [
+        createOwnCloudFileDetailProperty('fileid', true),
+      ]
+    );
+
+    const detail = folderFileDetails
+    .find(detail => detail.name === fileOrFolderName);
+
+    const fileid = detail.extraProperties['fileid'] as number;
+    result = await self.getObjectCreator(fileid);
+  } catch {}
+
+  return result || Promise.reject('Unable to find the creator.');
```

Comment:
What do you think of using a proper error type here instead of stringly-typed errors?